### DISCUSSION TITLE
Find a commit that includes a specific file

### DISCUSSION THREAD

**Discussion by BatmanAoD** [jj-vcs/jj#3047] [Q&A] [answered]
I have somehow lost a file that I added to my workspace. I have the `mv` command in my shell history showing the name that I gave it, but the file is not present in any of the commits I've checked. I also tried restoring a few operations from the op log, and still couldn't find it.

It's possible that I somehow deleted the file without `jj` ever getting the chance to snapshot it, but I'm hoping that's not the case.

Is there a way to search all commits for the existence of a particular file?

**Comment by martinvonz** [comment, marked as answer]
You can do `jj log -r .. <path>` to find any visible commits that touched the file. We unfortunately don't yet have a way of including all commits that have ever been visible (#2623). If you're in a colocated repo, you can probably do `git log --all <path>` to find it, since that will include all commits reachable from the `refs/jj/keep/*` refs.

**Reply by BatmanAoD** [reply]
THAT FOUND IT! Thank you so much.
